### ISSUE TITLE
NegativeArraySizeException when decoding some msgpack strings 

### ISSUE THREAD

**Issue by jamesnvc** [cognitect/transit-clj#33]
It seems that if I've encoded a string using msgpack with a length greater than or equal to 128 bytes and stored the header + length as the bytes `D9 80`, transit-clj gives a NegativeArraySizeException (stack trace at bottom).  If the header + length is instead provided in a two-byte format `DA 00 80`, the message is decoded as expected.

To reproduce, I generated a file with the following code:

```clojure
(require '[clojure.java.io :as io])
(require '[cognitect.transit :as transit])

(with-open [f (io/output-stream "test_good.msgpack")]
   (transit/write (transit/writer f :msgpack) (clojure.string/join "" (repeat 128 "X"))))
```

I then copied `test_good.msgpack` to `test_bad.msgpack`, opened it with a hex editor, and changed the bytes `DA 00 80` to `D9 80`.  I can then see that, while both files decode the same when using msgpack, trying to read the second file fails:

```clojure
(require '[clojure.java.io :as io])
(require '[cognitect.transit :as transit])

(defn read-transit [f] 
   (with-open [s (io/input-stream f)] 
      (transit/read (transit/reader s :msgpack))))

(read-transit "test_good.msgpack") ; => okay, string of 128 Xs
(read-transit "test_bad.msgpack") ; => exception
```

Examples of said files attached: [test files.zip](https://github.com/cognitect/transit-clj/files/382408/test.files.zip)

Verified msgpack identicalness with the following (using [clojure-msgpack 1.2.0](https://github.com/edma2/clojure-msgpack) ):

```clojure
(require '[clojure.java.io :as io])
(require '[msgpack.core :as msg])

(defn read-msg [fname]
   (with-open [f (io/input-stream fname)] (msg/unpack f)))

(= (read-msg "test_good.msgpack") (read-msg "test_bad.msgpack")) ; => true
```

Stack trace when trying to decode:

```
#error {
 :cause nil
 :via
 [{:type java.lang.RuntimeException
   :message "java.lang.NegativeArraySizeException"
   :at [com.cognitect.transit.impl.ReaderFactory$ReaderImpl read "ReaderFactory.java" 119]}
  {:type java.lang.NegativeArraySizeException
   :message nil
   :at [org.msgpack.unpacker.MessagePackUnpacker readRawBody "MessagePackUnpacker.java" 362]}]
 :trace
 [[org.msgpack.unpacker.MessagePackUnpacker readRawBody "MessagePackUnpacker.java" 362]
  [org.msgpack.unpacker.MessagePackUnpacker readOneWithoutStackLarge "MessagePackUnpacker.java" 228]
  [org.msgpack.unpacker.MessagePackUnpacker readOneWithoutStack "MessagePackUnpacker.java" 139]
  [org.msgpack.unpacker.MessagePackUnpacker readValue "MessagePackUnpacker.java" 566]
  [org.msgpack.unpacker.AbstractUnpacker readValue "AbstractUnpacker.java" 65]
  [com.cognitect.transit.impl.MsgpackParser parseVal "MsgpackParser.java" 55]
  [com.cognitect.transit.impl.MsgpackParser parseArray "MsgpackParser.java" 135]
  [com.cognitect.transit.impl.MsgpackParser parseVal "MsgpackParser.java" 53]
  [com.cognitect.transit.impl.MsgpackParser parse "MsgpackParser.java" 44]
  [com.cognitect.transit.impl.ReaderFactory$ReaderImpl read "ReaderFactory.java" 117]
  [cognitect.transit$read invokeStatic "transit.clj" 296]
  [cognitect.transit$read invoke "transit.clj" 293]
```

**Comment by jamesnvc** [comment]
Caused by https://github.com/msgpack/msgpack-java/issues/213

**Comment by puredanger** [comment]
This looks like https://github.com/msgpack/msgpack-java/issues/213 which was fixed in msgpack 0.6.12. We might also want to consider a newer version than that as a dep as latest is 0.8.8

**Comment by puredanger** [comment]
I've released new versions of transit-java (depending on msgpack 0.6.12) and transit-clj (0.8.288) that uses it, which should resolve this particular problem. Updating to newer (0.8.x) version of msgpack will be considered at some point (it require more code changes due to API mods).
